commons/src/main/java/fr/free/nrw/commons/upload/GPSExtractor.java

```diff
@@ -33,21 +52,73 @@ public String getCoords() {
         }
 
         if (exif.getAttribute(ExifInterface.TAG_GPS_LATITUDE) == null) {
-            Log.d("Image", "Picture has no GPS info");
-            return null;
-        }
-        else {
+            imageCoordsExists = false;
+            Log.d(TAG, "Picture has no GPS info");
+
+            //Check what user's preference is for automatic location detection
+            boolean gpsPrefEnabled = gpsPreferenceEnabled();
+
+            if (gpsPrefEnabled) {
+                //If pref enabled, set up LocationListener to get current location
+                LocationManager locationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+                Criteria criteria = new Criteria();
+                String provider = locationManager.getBestProvider(criteria, true);
+
+                myLocationListener = new MyLocationListener();
+                locationManager.requestLocationUpdates(provider, 400, 1, myLocationListener);
+                Location location = locationManager.getLastKnownLocation(provider);
+
+                if (location != null) {
```

Comment:
The location is quite likely to be unavailable at this time if you literally only requested updates a few statements ago. Consider requesting a location update in onResume() and unregistering for updates in onPause.